src/spritekit.cs

```diff
@@ -1464,23 +1528,65 @@ interface SKUniform : NSCopying, NSCoding {
 		[Export ("initWithName:float:")]
 		IntPtr Constructor (string name, float /* float, not CGFloat */ value);
 
+		[Internal]
+		[Availability (Deprecated = Platform.iOS_10_0 | Platform.Mac_10_12)]
 		[Export ("initWithName:floatVector2:")]
-		IntPtr Constructor (string name, Vector2 value);
+		IntPtr InitWithNameFloatVector2 (string name, Vector2 value);
+
+		[Internal]
+		[iOS (10,0)][Mac (10,12)]
+		[Export ("initWithName:vectorFloat2:")]
+		IntPtr InitWithNameVectorFloat2 (string name, Vector2 value);
```

Comment:
You're assuming that *native* `GLKVector2` == *native* `vector_float2` == *managed* `Vector2`

    -+ (instancetype)uniformWithName:(NSString *)name floatVector2:(GLKVector2)value;
    ++ (instancetype)uniformWithName:(NSString *)name vectorFloat2:(vector_float2)value NS_AVAILABLE(10_12, 10_0);

That's likely why you get errors. @rolfbjarne added support for SIMD using

	[MarshalDirective (NativePrefix = "xamarin_simd__", Library = "__Internal")]

and it might cover this case (for the new API) but it need some tweaks.